cloudpub/ms_azure/utils.py

```diff
@@ -235,13 +262,25 @@ def prepare_vm_images(
         return [VMImageDefinition.from_json(json_gen1)]
 
 
+def _len_vm_images(disk_versions: List[DiskVersion]) -> int:
+    count = 0
+    for disk_version in disk_versions:
+        count = count + len(disk_version.vm_images)
+    return count
+
+
 def _build_skus(
     disk_versions: List[DiskVersion],
     default_gen: str,
     alt_gen: str,
     plan_name: str,
     security_type: Optional[List[str]] = None,
 ) -> List[VMISku]:
+    def get_skuid(arch):
+        if arch == "x64":
+            return plan_name
+        return f"{plan_name}-{arch.lower()}"
```

Comment:
@lslebodn just a brief explanation on how the `update_skus` works with this PR:

1. If no SKU is set it will parse all disk images and create the SKUs from scratch
2. If the number of SKUs match the number of disk images it will return them, as we can't re-generate those SKU IDs because it will cause the `PlanID property is locked` issue with the Graph API.
3. If at least one SKU is missing it will add the missing one(s) giving a unique name for it (them).

Naming:
1. If the plan contains x86 images and Gen2 is default:
    |SKU ID  |Image Type    |Default|
    |------------|--------------------|----------|
    |{planid}|x64 Gen2 | ✔        |
    |{planid}-gen1|x64 Gen1 | |
    |{planid}-{arch}| arm64 Gen2 | |

2. If the plan contains x86 images and Gen1 is default:
    |SKU ID  |Image Type    |Default|
    |------------|--------------------|----------|
    |{planid}|x64 Gen1 | ✔        |
    |{planid}-gen2|x64 Gen2 | |
    |{planid}-{arch}-gen2| arm64 Gen2 | |


2. If the plan contains only arm64 images:
    |SKU ID  |Image Type    |Default|
    |------------|--------------------|----------|
    |{planid}-{arch}|arm64 Gen2 | ✔        |


With this I don't think we need to worry much about having it configurable as it will always generate a unique name for each SKU and reuse the existing ones when available.